edi_oca/models/edi_configuration.py

```diff
@@ -0,0 +1,189 @@
+# Copyright 2024 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import datetime
+
+import pytz
+
+from odoo import api, fields, models
+from odoo.tools import DotDict, safe_eval
+
+
+def date_to_datetime(dt):
+    """Convert date to datetime."""
+    if isinstance(dt, datetime.date):
+        return datetime.datetime.combine(dt, datetime.datetime.min.time())
+    return dt
+
+
+def to_utc(dt):
+    """Convert date or datetime to UTC."""
+    # Gracefully convert to datetime if needed 1st
+    return date_to_datetime(dt).astimezone(pytz.UTC)
+
+
+class EdiConfiguration(models.Model):
+    _name = "edi.configuration"
+
+    name = fields.Char(string="Name")
+    active = fields.Boolean(default=True)
+    code = fields.Char(required=True, copy=False, index=True, unique=True)
+    description = fields.Char(help="describe what the conf is for")
+    backend_id = fields.Many2one(string="Backend", comodel_name="edi.backend")
+    type_id = fields.Many2one(string="Exchange Type", comodel_name="edi.exchange.type")
+    model = fields.Many2one(
+        "ir.model", string="Model", help="model the conf applies to"
+    )
+    res_id = fields.Many2one(
+        "res.partner", string="Partner", ondelete="cascade", required=False
+    )
```

Comment:
I think we have some mistakes, why `res_id` here only references to `res.partner`?

I can't get a whole context for a single record here, maybe will discuss later